pkg/service/restore/schema_worker.go

```diff
@@ -273,3 +323,93 @@ func (w *schemaWorker) getFileNamesMapping(sstables []string, sstableUUIDFormat
 	}
 	return sstable.RenameToIDs(sstables, &w.generationCnt)
 }
+
+func getDescribedSchema(ctx context.Context, client *scyllaclient.Client, snapshotTag string, locHost map[Location][]string) (*query.DescribedSchema, error) {
+	baseDir := path.Join("backup", string(SchemaDirKind))
+	// It's enough to get a single schema file, but it's important to validate
+	// that each location contains exactly one or none of them.
```

Comment:
Should they be compared against the equal content or hash ? Is the consistency guaranteed by awaiting schema agreement ?

Reply:
Too be honest, this check could be removed.

This check is about verifying if user specified correct locations - they either all have schema file with given snapshot tag, or none of them have it. It could be extended to checking file hash, but IMO this would be an overkill (perhaps this check is already a one). 

In general, when performing backup, we ask only a single node for the DESC SCHEMA and we backup it to all specified backup locations. This means, that we don't store per node DESC SCHEMA output, as it would be difficult to use it during restore. It also means, that all locations have the same file content.

In terms of schema agreement, it turns out that we don't even need to have it when querying DESC SCHEMA (see https://github.com/scylladb/scylladb/issues/19213 for more details), but I have yet to change that in another commit.

Reply:
So I'm fine with removing this check or just leaving it as it is. What is your opinion on this matter?

Reply:
I'm fine with leaving it as is.